LanguageFeatures/Patterns/type_inference_A08_t03.dart

```diff
@@ -51,10 +53,16 @@ main() {
     v2.expectStaticType<Exactly<B>>();
     v3.expectStaticType<Exactly<C>>();
   }
+  String log = "";
   try {
-    final {"key1": A v1, "key2": B v2, "key3": v3} = {};
-    v1.expectStaticType<Exactly<A>>();
-    v2.expectStaticType<Exactly<B>>();
-    v3.expectStaticType<Exactly<B>>();
+    var {"key1": A v1, "key2": B v2, "key3": v3} =
+        getType({}, (String s) {log += s;});
+  } catch (_) {}
+  Expect.equals("Map<Object?, B>", log);
```

Comment:
Is that key type actually going to be `Object?`? Checking the spec: [Yes!](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=are%20_.-,Else%20K%20is%20_,-and%20V%20is)

This is slightly surprising (it seems likely that this will give the map itself a less informative run-time type in some cases, including the one given here).